modules/n1ql/pages/n1ql-language-reference/dropsequence.adoc

```diff
@@ -0,0 +1,122 @@
+= DROP SEQUENCE
+:page-topic-type: reference
+:page-status: Couchbase Server 7.6
+:imagesdir: ../../assets/images
+:description: The DROP SEQUENCE statement enables you to drop a sequence in a given keyspace.
+
+// Cross-references
+:authorization-overview: xref:learn:security/authorization-overview.adoc
+:query-context: xref:n1ql:n1ql-intro/queriesandresults.adoc#query-context
+:identifiers: xref:n1ql-language-reference/identifiers.adoc
+:logical-hierarchy: xref:n1ql-intro/sysinfo.adoc#logical-hierarchy
+
+// TEMP
+include::partial$n1ql-language-reference/horizontal-style.adoc[]
+
+[abstract]
+{description}
+
+== Purpose
+
+include::./sequenceops.adoc[tags=overview]
+
+== Prerequisites
+
+.RBAC Privileges
+To execute the DROP SEQUENCE statement, you must have the _Query Manage Sequences_ privilege granted on the scope.
+For more details about user roles, see {authorization-overview}[Authorization].
+
+== Syntax
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=drop-sequence]
+----
+
+image::n1ql-language-reference/drop-sequence.png["Syntax diagram: refer to source code listing", align=left]
+
+// TODO: Automatic links in EBNF.
+
+[horizontal]
+sequence:: (Required) A name that identifies the sequence within a namespace, bucket, and scope.
+See <<sequence>> below.
+
+[[sequence]]
+=== Sequence Name
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=sequence]
+----
+
+image::n1ql-language-reference/sequence.png["Syntax diagram: refer to source code listing", align=left]
+
+The sequence name specifies the name of the sequence to drop.
+
+Each sequence is associated with a given namespace, bucket, and scope.
+You must specify the namespace, bucket, and scope to refer to the sequence correctly.
+
+[horizontal]
+namespace::
+(Optional) The {logical-hierarchy}[namespace] of the bucket which contains the sequence you want to drop.
+
+bucket::
+(Optional) The bucket which contains the sequence you want to drop.
+
+scope::
+(Optional) The scope which contains the sequence you want to drop.
+
+identifier::
+(Required) The name of the sequence.
+The sequence name is case-sensitive.
+
+Currently, only the `default` namespace is available.
+If you omit the namespace, the default namespace in the current session is used.
+
+If the {query-context}[query context] is set, you can omit the bucket and scope from the statement.
```

Comment:
Would this not be "you can omit leading components of the name that are resolved via the query_context" ? - Since you can set the query context to a bucket and then provide just scope.name to reference a sequence (same as for keyspaces etc.).

Reply:
Interesting! I don't think we currently document that behavior. Particularly as the query context control in the GUI doesn't allow you to select a bucket without selecting a scope. This is something we'd need to look at throughout the Query docs rather than just here. Will leave as is for now but might raise a ticket to look at this.